client/core/core.go

```diff
@@ -1065,6 +1075,62 @@ func (c *Core) handleUnbookOrderMsg(dc *dexConnection, msg *msgjson.Message) err
 	return ob.Unbook(&note)
 }
 
+// handleEpochOrderMsg is called when an epoch_order notification is
+// received.
+func (c *Core) handleEpochOrderMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.EpochOrderNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("epoch order note unmarshal error: %v", err)
+	}
+
+	dc.epochsMtx.Lock()
+	defer dc.epochsMtx.Unlock()
+
+	ep, ok := dc.epochs[note.MarketID]
+	if !ok {
+		return fmt.Errorf("no epoch queue found with market id %q",
+			note.MarketID)
+	}
+
+	return ep.Enqueue(&note)
+}
+
+// handleMatchProofMsg is called when a match_proof notification is received.
+func (c *Core) handleMatchProofMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.MatchProofNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("match proof note unmarshal error: %v", err)
+	}
+
+	dc.epochsMtx.Lock()
+	defer dc.epochsMtx.Unlock()
+
+	ep, ok := dc.epochs[note.MarketID]
+	if !ok {
+		return fmt.Errorf("no epoch queue found with market id %q",
+			note.MarketID)
+	}
+
+	seed, csum, err := ep.GenerateMatchProof(note.Preimages, note.Misses)
+	if err != nil {
+		return fmt.Errorf("unable to generagte match proof %v", err)
+	}
+
+	if !bytes.Equal(seed, note.Seed) {
+		return fmt.Errorf("match proof seed mismatch: expected %x, got %x",
+			note.Seed, seed)
+	}
+
+	if !bytes.Equal(csum, note.CSum) {
+		return fmt.Errorf("match proof csum: expectd %x, got %x",
```

Comment:
expectd
Also, the other check uses the term "mismatch", so might as well here too.